### PR TITLE
Add workflow to test Airflow RC

### PR DESCRIPTION
**Pull request by pankajastro**
closes: https://github.com/astronomer/astro-sdk/issues/1950
mostly taken from https://github.com/astronomer/astronomer-providers

### CODE REVIEW

**Review by tatiana — APPROVED** [review]
This will help us avoid issues like we had with the recent MS provider release, thanks @pankajastro !

**Review by utkarsharma2 — APPROVED** [review]

**Review by Lee-W — APPROVED** [review]
I'm good with it now. But I think we should make it reusable later. Let me see how we can do that. (Also found a typo by me 🫠 but I think that's something we should fix on the provider side when we make it resuable)

File: python-sdk/dev/integration_test_scripts/replace_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        if requirement.specifier:
+            pinned_package = f"{package_name_to_search}{requirement.specifier}"
+        elif requirement.url:
+            pinned_package = f"{package_name_to_search} @{requirement.url}"
+        else:
+            raise Exception(
+                f"Invalid package {package} provided. It needs to be pinned to a specific version."
+            )
+
+        with fileinput.FileInput("pyproject.toml", inplace=True) as setup_file:
```
Comment (pankajkoti) [inline python-sdk/dev/integration_test_scripts/replace_dependencies.py:68]: Were we able to test this workflow on a RC url? I am wondering if the path works here or it needs to be `python_sdk/pyproject.toml`

Comment (pankajastro) [reply python-sdk/dev/integration_test_scripts/replace_dependencies.py:68]: thanks for pointing it out. I have added `working-directory: python-sdk` hopefully should work

**Pull request merged**